source/jormungandr/tests/kirin_realtime_tests.py

```diff
@@ -2139,6 +2139,231 @@ def test_add_new_trip_present_in_navitia(self):
+            effect='delayed',
+        )
+
+        # The disruption created above is modified so no disruption is added
+        disruptions_after = self.query_region(disruption_query)
+        assert initial_nb_disruptions + 1 == len(disruptions_after['disruptions'])
+
+        # Again we have 8 vehicle_journeys
+        pt_response = self.query_region('vehicle_journeys')
+        assert initial_nb_vehicle_journeys + 1 == len(pt_response['vehicle_journeys'])
+
+        # /journeys from B to A with departure at 20120615T003000 and arrival at 20120614T230000
+        journey_query = (
+            "journeys?from={f}&to={to}&data_freshness=realtime&"
+            "datetime={dt}&_current_datetime={dt}&direct_path=none&walking_speed=0.001".format(
+                f='stop_point:stopC', to='stop_point:stopA', dt='20120614T230000'
+            )
+        )
+        response = self.query_region(journey_query)
+        assert len(response['journeys']) == 1
+        assert response['journeys'][0]['departure_date_time'] == '20120615T003000'
+        assert response['journeys'][0]['arrival_date_time'] == '20120615T010102'
+
+
+@dataset(MAIN_ROUTING_TEST_SETTING)
+class TestKirinDelayPassMidnightTowardsPreviousDay(MockKirinDisruptionsFixture):
+    def test_delay_pass_midnight_towards_previous_day(self):
+        """
+        Relates to "test_cots_update_trip_with_advance_pass_midnight_on_first_station" in kirin
+        1. Verify disruption count, vehicle_journeys count and journey
+        2. Add a disruption with a delay in first station (stop_point:stopB) so that there is a pass midnight
+        towards previous day
+        3. Verify disruption count, vehicle_journeys count and journey
+        """
+        disruption_query = 'disruptions?_current_datetime={dt}'.format(dt='20120614T080000')
+        initial_nb_disruptions = len(self.query_region(disruption_query)['disruptions'])
+
+        # We have 7 vehicle_journeys at the beginning
+        pt_response = self.query_region('vehicle_journeys')
+        initial_nb_vehicle_journeys = len(pt_response['vehicle_journeys'])
+        assert initial_nb_vehicle_journeys == 7
+
+        # /journeys from B to A with departure at 20120614T080100 and arrival at 20120614T080102
+        journey_query = (
+            "journeys?from={f}&to={to}&data_freshness=base_schedule&"
+            "datetime={dt}&_current_datetime={dt}&direct_path=none&walking_speed=0.001".format(
+                f='stop_point:stopB', to='stop_point:stopA', dt='20120615T080000'
+            )
+        )
+        response = self.query_region(journey_query)
+        assert len(response['journeys']) == 1
+        assert response['journeys'][0]['departure_date_time'] == '20120615T080100'
+        assert response['journeys'][0]['arrival_date_time'] == '20120615T080102'
+
+        # A new disruption with a detour on departure station to have  a pass midnight in advance
+        self.send_mock(
+            "vjB",
+            "20120614",
```

Comment:
This may be the wrong date... Related to the discussion in Kirin: https://github.com/CanalTP/kirin/pull/242/files#r285073475